src/MongoDB.Bson/Serialization/Conventions/ImmutableTypeClassMapConvention.cs

```diff
@@ -35,8 +38,7 @@ public void Apply(BsonClassMap classMap)
                 return;
             }
 
-            var propertyBindingFlags = BindingFlags.Public | BindingFlags.Instance;
-            var properties = typeInfo.GetProperties(propertyBindingFlags);
+            var properties = typeInfo.GetProperties(PropertyBindingsFlags);
```

Comment:
I think the entire bug can be fixed with just these few lines:

    var propertyBindingFlags = BindingFlags.Public | BindingFlags.Instance;
    var properties =
        typeInfo.GetProperties(propertyBindingFlags)
        .Where(p => !p.GetCustomAttributes(typeof(BsonIgnoreAttribute)).Any())
        .ToArray();

The essence of the bug was that `[BsonIgnore]` wasn't being respected by the `ImmutableTypeClassMapConvention`.

Look into it and see what you think.

Reply:
I used the following minimal repro to test the one line change above:

    namespace TestCSharp2984
    {
        public abstract class B
        {
            [BsonIgnore]
            public abstract int A { get; }
        }
    
        public class C : B
        {
            public C(int p) { P = p; }
    
            [BsonIgnore]
            public override int A => 1;
            public int P { get; }
        }
    
        public static class Program
        {
            public static void Main(string[] args)
            {
                var c = new C(1);
                var json = c.ToJson();
            }
        }
    }